core/src/main/java/org/geysermc/geyser/command/defaults/ConnectionTestCommand.java

```diff
@@ -25,86 +25,86 @@
         if (!geyser.getConfig().getBedrock().address().equals("0.0.0.0")) {
-            sender.sendMessage("The address specified in `bedrock` `address` is not \"0.0.0.0\" - this may cause issues unless this is deliberate and intentional.");
+            source.sendMessage("The address specified in `bedrock` `address` is not \"0.0.0.0\" - this may cause issues unless this is deliberate and intentional.");
         }
 
         // Issue: did someone turn on enable-proxy-protocol and they didn't mean it?
         if (geyser.getConfig().getBedrock().isEnableProxyProtocol()) {
-            sender.sendMessage("You have the `enable-proxy-protocol` setting enabled. " +
+            source.sendMessage("You have the `enable-proxy-protocol` setting enabled. " +
                     "Unless you're deliberately using additional software that REQUIRES this setting, you may not need it enabled.");
         }
 
         CompletableFuture.runAsync(() -> {
             try {
                 // Issue: SRV record?
-                String ip = fullAddress[0];
-                String[] record = WebUtils.findSrvRecord(geyser, ip);
-                if (record != null && !ip.equals(record[3]) && !record[2].equals(String.valueOf(port))) {
-                    sender.sendMessage("Bedrock Edition does not support SRV records. Try connecting to your server using the address " + record[3] + " and the port " + record[2]
+                String[] record = WebUtils.findSrvRecord(geyser, address);
+                if (record != null && !address.equals(record[3]) && !record[2].equals(String.valueOf(port))) {
+                    source.sendMessage("Bedrock Edition does not support SRV records. Try connecting to your server using the address " + record[3] + " and the port " + record[2]
                             + ". If that fails, re-run this command with that address and port.");
                     return;
                 }
 
                 // Issue: does Loopback need applying?
                 if (LoopbackUtil.needsLoopback(GeyserImpl.getInstance().getLogger())) {
-                    sender.sendMessage("Loopback is not applied on this computer! You will have issues connecting from the same computer. " +
+                    source.sendMessage("Loopback is not applied on this computer! You will have issues connecting from the same computer. " +
                             "See here for steps on how to resolve: " + "https://wiki.geysermc.org/geyser/fixing-unable-to-connect-to-world/#using-geyser-on-the-same-computer");
                 }
 
                 // mcsrvstatus will likely be replaced in the future with our own service where we can also test
                 // around the OVH workaround without worrying about caching
-                JsonNode output = WebUtils.getJson("https://api.mcsrvstat.us/bedrock/2/" + args[0]);
+                JsonNode output = WebUtils.getJson("https://api.mcsrvstat.us/bedrock/2/" + address);
```

Comment:
this looks broken - `args[0]` was `address:port`

Reply:
But we do want to send both the address and port to test?
e.g. https://api.mcsrvstat.us/bedrock/2/test.geysermc.org:18382 would test `test.geysermc.org` with the port `18382` 

Not that it really matters, since #3951 re-works the checking to use the Geyser ping service that can also check for e.g. the OVH firewall issue

Reply:
I agree, but `address` does not include the port

Reply:
But it should, that's the point

Reply:
I'm talking about the `address` variable, you're talking about the endpoint.